### PR TITLE
format Python files using black extension

### PR DESCRIPTION
**Pull request by afeld**
~~This does a few things:~~

- ~~Sets the default formatter to be Prettier, which will also apply it to things like Markdown.~~
- Installs the Black extension to enable formatting Python files on save.
- ~~Formats on save by default, which is fine because formatting HTML is still disabled, protecting the Jinja templates.~~
- ~~Formats the settings files (with Prettier).~~

The rest has been moved to https://github.com/cal-itp/benefits/pull/746.

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -34,6 +22,7 @@
     "eamodio.gitlens",
     "mhutchie.git-graph",
     "ms-python.python",
+    "ms-python.black-formatter",
```
Comment (thekaveman) [inline .devcontainer/devcontainer.json:25]: What is the difference between installing the extension and [installing `black` directly](https://github.com/cal-itp/benefits/blob/dev/.devcontainer/Dockerfile#L16) as part of the devcontainer build? My understanding is the `black` formatting has already been working for a while now...

File: .vscode/settings.json [hunk 2]
```diff
@@ -13,24 +8,11 @@
...
   "files.trimTrailingWhitespace": true,
-  "[javascript]": {
-    "editor.formatOnSave": true,
-    "editor.defaultFormatter": "esbenp.prettier-vscode"
-  },
-  "[json]": {
-    "editor.formatOnSave": true,
-    "editor.defaultFormatter": "esbenp.prettier-vscode"
-  },
   "python.formatting.provider": "black",
```
Comment (thekaveman) [inline .vscode/settings.json:11]: I'm pretty sure this is what tells VS Code to use `black` - see comment above.

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (afeld) [reply .devcontainer/devcontainer.json:25]: `black` is installed as a package, but wasn't actually being run from the editor. The extension does the latter. Steps to reproduce, without this pull request:

1. Modify a Python file in a way that should be undone with formatting (adding whitespace or whatever).
1. Save — the file doesn't change.
1. From the command palette, run `Format Document`.

   Screen Shot 2022-06-13 at 1 18 18 PM

1. This dialog appears:

   Screen Shot 2022-06-13 at 1 18 26 PM

File: .vscode/settings.json (hunk 2 above)
Comment (afeld) [reply .vscode/settings.json:11]: Correct, but it appears the extension is needed as well.

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (thekaveman) [reply .devcontainer/devcontainer.json:25]: I don't see this behavior on the existing `dev`

1. Modify a Python file
2. Save - the file is formatted with `black`

save-black

Comment (afeld) [reply .devcontainer/devcontainer.json:25]: Weird. Can you confirm you don't have the `black` extension installed? @machikoyasuda @angela-tran Do you get formatting on save, or no?

Comment (thekaveman) [reply .devcontainer/devcontainer.json:25]: Confirmed

image

Comment (angela-tran) [reply .devcontainer/devcontainer.json:25]: @afeld Formatting on save has been working for me as well. I wonder if there might be some difference with Macs?

Comment (afeld) [reply .devcontainer/devcontainer.json:25]: Mind trying rebuilding on this branch? If that works for you all, mind if we add the extension, even if it only makes a difference for me (for some reason)?

Comment (angela-tran) [reply .devcontainer/devcontainer.json:25]: There are some troubleshooting tips here: https://code.visualstudio.com/docs/python/editing#_troubleshooting-formatting

Comment (angela-tran) [reply .devcontainer/devcontainer.json:25]: I can also rebuild on your branch and make sure that having the extension would be ok

**Comment by thekaveman** [comment]
Let's remove `black` from the devcontainer build process and just use the extension since it seems to work for more cases!

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (angela-tran) [reply .devcontainer/devcontainer.json:25]: I followed these steps:


1. Modify a Python file in a way that should be undone with formatting (adding whitespace or whatever).
1. Save — the file doesn't change.
1. From the command palette, run `Format Document`.
1. I get a different dialog
image

When I click "Configure" and choose "Black Formatter", this is added to `.vscode/settings.json`, and code is formatted on save:
```
  "[python]": {
    "editor.defaultFormatter": "ms-python.black-formatter"
  }
```

Comment (afeld) [reply .devcontainer/devcontainer.json:25]: Asked for clarification: https://github.com/microsoft/vscode-black-formatter/issues/82

**Review by thekaveman — COMMENTED** [review]
I'm now getting this error message when I _Rebuild and Reopen_ the devcontainer on this branch:

image

I'm also _not_ getting Format on Save anymore:

no-format-on-save

**Comment by afeld** [comment]
Good gracious. Asked in https://github.com/microsoft/vscode-black-formatter/issues/82#issuecomment-1155478052. Does format on save work after you click `Install and Reload`?

**Comment by thekaveman** [comment]
Going back to draft while this is sorted out to remove from the review queue.

**Comment by afeld** [comment]
Split this pull request into two to hopefully allow the other changes to get in.

How are we feeling about this pull request? [The `Install and Reload` problem is a broader bug](https://github.com/microsoft/vscode/issues/149309); dunno if that's a deal-breaker for folks or not. Again, I'm not writing enough Python on this project that having to install the extension manually is a huge deal for me, so if everyone else is happy with the current setup, fine to just close this.

File: .vscode/settings.json [hunk 3]
```diff
@@ -8,10 +8,11 @@
   "files.insertFinalNewline": true,
   "files.trimFinalNewlines": true,
   "files.trimTrailingWhitespace": true,
+  // https://github.com/microsoft/vscode-black-formatter/issues/82#issuecomment-1155492426
```
Comment (afeld) [inline .vscode/settings.json:11]: [FYI that this file supports comments.](https://code.visualstudio.com/Docs/languages/json#_json-with-comments)

**Comment by afeld** [comment]
On second thought, since I got formatting Python on save to work without the extra extension, this doesn't add value, and having to `Install and Reload` is annoying.

**Pull request closed without merging**